tests/org.eclipse.sirius.emfjson.tests/src/main/java/org/eclipse/sirius/emfjson/tests/internal/integration/JsonSerializationAPITests.java

```diff
@@ -62,4 +86,138 @@ public void serializeResourceToJsonElementTest() {
 
         Assert.assertEquals(expectedResource, jsonTreeString);
     }
+
+    /**
+     * Check that the deserialization of a mono and multivalued reference are correctly done so that the proxy can be
+     * properly resolved.
+     */
+    @Test
+    public void serializeInterResourceReferenceTest() {
+        this.options.put(JsonResource.OPTION_DISPLAY_DYNAMIC_INSTANCES, Boolean.TRUE);
+
+        JsonResourceImpl resource1 = new JsonResourceImpl(URI.createURI("sirius:///resource1"), this.options); //$NON-NLS-1$
+        JsonResourceImpl resource2 = new JsonResourceImpl(URI.createURI("sirius:///resource2"), this.options); //$NON-NLS-1$
+        this.initializeResource(resource1, resource2);
+
+        // get the json of the resource1
+        String jsonRes1 = ""; //$NON-NLS-1$
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        try {
+            resource1.save(outputStream, this.options);
+
+            byte[] bytes = outputStream.toByteArray();
+            jsonRes1 = new String(bytes);
+
+        } catch (IOException e) {
+            Assert.fail(e.getMessage());
+        } finally {
+            try {
+                outputStream.close();
+            } catch (IOException e) {
+                Assert.fail(e.getMessage());
+            }
+        }
+
+        // get the json of the resource2
+        String jsonRes2 = ""; //$NON-NLS-1$
+        outputStream = new ByteArrayOutputStream();
+        try {
+            resource2.save(outputStream, this.options);
+
+            byte[] bytes = outputStream.toByteArray();
+            jsonRes2 = new String(bytes);
+
+        } catch (IOException e) {
+            Assert.fail(e.getMessage());
+        } finally {
+            try {
+                outputStream.close();
+            } catch (IOException e) {
+                Assert.fail(e.getMessage());
+            }
+        }
```

Comment:
I did not known how to create the json file at the beggining. But now may as well keep the code and thus it will be easy to update it if needed.

Reply:
I prefer this test uses files to load the context (like the other tests). Moreover, the two files will rather be small. I think it will be easier to maintain.